apps/antalmanac/src/components/Header/SettingsMenu.tsx

```diff
@@ -141,11 +144,26 @@ function TimeMenu() {
 
 function ExperimentalMenu() {
     const [previewMode, setPreviewMode] = usePreviewStore((store) => [store.previewMode, store.setPreviewMode]);
+    const [autoSave, setAutoSave] = useAutoSaveStore((store) => [store.autoSave, store.setAutoSave]);
 
     const handlePreviewChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         setPreviewMode(event.target.checked);
     };
 
+    const handleAutoSaveChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        setAutoSave(event.target.checked);
+        if (event.target.checked) {
+            const savedUserID = window.localStorage.getItem('userID');
+
+            if (savedUserID) {
+                actionTypesStore.emit('autoSaveStart');
+                await saveSchedule(savedUserID, true, true);
+                appStore.unsavedChanges = false;
+                actionTypesStore.emit('autoSaveEnd');
+            }
+        }
```

Comment:
Use guard clauses to reduce nesting.
```suggestion
        if (!event.target.checked) return
        
        const savedUserId = window.localStorage.getItem('userID');
        
        if (!savedUserId) return

        actionTypesStore.emit('autoSaveStart');
        await saveSchedule(savedUserID, true, true);
        appStore.unsavedChanges = false;
        actionTypesStore.emit('autoSaveEnd');
```